components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -925,6 +930,112 @@ public UserRecoveryData getUserRecoveryData(String code, RecoverySteps step) thr
         return recoveryData;
     }
 
+    /**
+     * Get user recovery data using the recovery flow id.
+     *
+     * @param recoveryFlowId Recovery flow id of the user.
+     * @param step Recovery step
+     * @throws IdentityRecoveryException If an error occurred while validating the recoveryId.
+     */
+    public UserRecoveryData getUserRecoveryDataFromFlowId(String recoveryFlowId, RecoverySteps step)
+            throws IdentityRecoveryException {
+
+        UserRecoveryData recoveryData;
+        UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
+        try {
+            // Retrieve recovery data bound to the recoveryFlowId.
+            recoveryData = userRecoveryDataStore.loadFromRecoveryFlowId(recoveryFlowId, step);
+        } catch (IdentityRecoveryException e) {
+            // Map code expired error to new error codes for user account recovery.
+            if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_FLOW_ID.getCode().equals(e.getErrorCode())) {
```

Comment:
Instead of doing the error code mapping here, can't we handle it from the place it is been thrown?

Reply:
This is thrown from the DAO layer and it can be used for multiple recovery scenarios. Hence, let's keep this to preserve the consistency.